crates/primitives/src/transaction/mod.rs

```diff
@@ -1100,10 +1101,38 @@ pub struct TransactionSigned {
     pub transaction: Transaction,
 }
 
+impl Clone for TransactionSigned {
+    fn clone(&self) -> Self {
+        Self {
+            hash: LazyLock::new(|| Default::default()),
```

Comment:
ah looks like we actually need OnceLock here because we need to call .get_or_init(||)